Document `LOAD ON ALL DATA` privilege (#737)
neo4j/docs-cypher: modules/ROOT/pages/administration/access-control/load-privileges.adoc

////
[source, cypher, role=test-setup]
----
CREATE ROLE roleLoadAllData
----
////

[role=enterprise-edition aura-db-enterprise]
[[access-control-load-privileges]]
= Load privileges

[abstract]
--
This section explains how to use Cypher to manage load privileges.
--

All load privileges are relevant system-wide.
Like DBMS privileges, they do not belong to one specific database or graph.
For more details on the differences between graphs, databases and the DBMS, refer to xref::introduction/cypher_neo4j.adoc[].

image::privileges_grant_and_deny_syntax_load_privileges.svg[title="Syntax of GRANT and DENY load Privileges"]

// TODO: add image later when there is more than one LOAD privilege
//image::privileges_hierarchy_load.svg[title="Load privileges hierarchy"]

The load privileges applies to the xref:clauses/load-csv.adoc[`LOAD CSV` clause], filtering whether the data can be loaded from the given source.

The load privileges are assignable using Cypher administrative commands.
They can be granted, denied and revoked like other privileges.

[NOTE]
====
More details about the syntax descriptions can be found xref:administration/index.adoc#administration-syntax[here].

> ====

Reviewer: I would prefer this note after the table. Otherwise people (like me) might follow the link before having read the basic syntax, at which point the syntax details aren't very useful

Author: I think it's generally before our syntax descriptions elsewhere but I can move it down 🤷

.Load privileges command syntax
[options="header", width="100%", cols="3a,2"]
|===
| Command | Description

| [source, syntax, role=noheader]
GRANT [IMMUTABLE] LOAD
  ON ALL DATA
  TO role[, ...]
| Enables the specified roles to load external data in queries.

|===

[[access-control-load-all-data]]
== The `ALL DATA` privilege

The load privilege on `ALL DATA` enables or disables loading data.
When granted the user can load data from any source and when missing or denied no data can be loaded at all.

For example, we can enable users with the role `roleLoadAllData` to be able to load data with `LOAD CSV`:

[source, cypher, role=noplay]
----
GRANT LOAD ON ALL DATA TO roleLoadAllData
----

List all privileges for the role `roleLoadAllData` as commands by using the following query:

[source, cypher, role=noplay]
----
SHOW ROLE roleLoadAllData PRIVILEGES AS COMMANDS
----

.Result
[options="header,footer", width="100%", cols="m"]
|===
|command
|"GRANT LOAD ON ALL DATA TO `roleLoadAllData`"
a|Rows: 1
|===

> 

Reviewer: Maybe a note here about PUBLIC having the privilege by default?

:description: How to use Cypher to manage load privileges.

////
[source, cypher, role=test-setup]
----
CREATE ROLE roleLoadAllData
----
////

[role=enterprise-edition aura-db-enterprise]
[[access-control-load-privileges]]
= Load privileges

> [abstract]
> --
> This section explains how to use Cypher to manage load privileges.
> --
> 

Reviewer 2: We don't use abstracts any more (I have been meaning to get to deleting all existing ones, but have not got around to it yet :/) 
```suggestion
_This feature was released in Neo4j 5.13._

This section explains how to use Cypher to manage load privileges.
```

> All load privileges are relevant system-wide.

Reviewer 2: My suggestion may not make sense, but I find "relevant system-wide" ambiguous.

```suggestion
All load privileges apply to the whole system.
```

Like DBMS privileges, they do not belong to one specific database or graph.

> For more details on the differences between graphs, databases and the DBMS, refer to xref::introduction/cypher_neo4j.adoc[].

Reviewer 2: ### 
```suggestion
For more details on the differences between graphs, databases, and the DBMS, refer to xref::introduction/cypher_neo4j.adoc[].
```

> image::privileges_grant_and_deny_syntax_load_privileges.svg[title="Syntax of GRANT and DENY load Privileges"]

Reviewer 2: Tried this locally and it looks fine.
```suggestion
image::privileges_grant_and_deny_syntax_load_privileges.svg[width="800", title="Syntax of GRANT and DENY load Privileges"]
```

// TODO: add image later when there is more than one LOAD privilege
//image::privileges_hierarchy_load.svg[title="Load privileges hierarchy"]

> The load privileges applies to the xref:clauses/load-csv.adoc[`LOAD CSV` clause], filtering whether the data can be loaded from the given source.

Reviewer 2: ```suggestion
The load privileges apply to the xref:clauses/load-csv.adoc[`LOAD CSV` clause], deciding whether or not the data can be loaded from the given source.
```

> The load privileges are assignable using Cypher administrative commands.

Reviewer 2: ```suggestion
The load privileges are assigned using Cypher administrative commands.
```

> They can be granted, denied and revoked like other privileges.

Reviewer 2: ```suggestion
They can be granted, denied, and revoked in the same way as other privileges.
For more details, see xref:administration/access-control/manage-privileges.adoc[].
```

.Load privileges command syntax
[options="header", width="100%", cols="3a,2"]
|===
| Command | Description

| [source, syntax, role=noheader]
GRANT [IMMUTABLE] LOAD
  ON ALL DATA
  TO role[, ...]
| Enables the specified roles to load external data in queries.

|===

[NOTE]
====
More details about the syntax descriptions can be found xref:administration/index.adoc#administration-syntax[here].
====

> Unlike other privileges these are not granted, denied, or revoked on `DBMS`, `Database`, or `Graph` but instead on the data source to load from.

Reviewer 2: ```suggestion
Unlike other privileges, the `LOAD` privilege is not granted, denied, or revoked on `DBMS`, `Database`, or `Graph`, but instead on the data source to load from.
```

> Currently, only on `ALL DATA` meaning all sources.

Reviewer 2: Are there concrete plans to change it? Otherwise, I would avoid "currently".
```suggestion
Adding `ON ALL DATA` means a role has the privilege to load data from all sources.
```

Author: Do we want to point out that it is (currently) the only alternative? or we skip that completely?

Reviewer 2: I think we skip. The docs tend to not make reference to future changes to the product.

[[access-control-load-all-data]]
== The `ALL DATA` privilege

The load privilege on `ALL DATA` enables or disables loading data.

> When granted the user can load data from any source and when missing or denied no data can be loaded at all.

Reviewer 2: ```suggestion
If granted, the user can load data from any source.
If missing or denied, no data can be loaded at all.
```

> For example, we can enable users with the role `roleLoadAllData` to be able to load data with `LOAD CSV`:

Reviewer 2: ```suggestion
For example, the below query grants users with the role `roleLoadAllData` the ability to load data with `LOAD CSV`:
```

[source, cypher, role=noplay]
----
GRANT LOAD ON ALL DATA TO roleLoadAllData
----

> List all privileges for the role `roleLoadAllData` as commands by using the following query:

Reviewer 2: ```suggestion
The below query lists all privileges for the role `roleLoadAllData` as commands:
```
